EIPS/eip-969.md

```diff
@@ -0,0 +1,167 @@
+## Rationale
+
+This EIP is aimed at breaking existing ASIC based miners via small changes to the
+existing ethash algorithm.  We hope to accomplish the following:
+
+1. Break existing ASIC based miners.
+2. Demonstrate a willingness to fork in the event of future ASIC miner production.
+
+Goal #1 is something that we can only do probabalistically without detailed
+knowledge of existing ASIC miner design.  Our approach should balance the
+inherent security risks involved with changing the mining algorithm with the
+risk that the change we make does not break existing ASIC miners.  This EIP
+leans towards minimizing the security risks by making minimal changes to the
+algorithm accepting the risk that the change may not break dedicated hardware 
+miners that utilize partially or fully configurable logic. 
+
+Furthermore, we do not wish to introduce significant algorithm changes that
+may alter the power utilization or performance profile of existing GPU hardware.
+
+The change of FNV constant is a minimal change that can be quickly
+implemented across the various network node and miner implementations.
+
+It is proposed that `ASIC_MITIGATION_FORK_BLKNUM` be no more than 5550000 (epoch 185), giving
+around 30 days of notice to node and miner developers and a sufficient window
+for formal analysis of the changes by experts. We must weigh this window against
+the risk introduced by allowing ASICs that may exist to continue to propagate
+on the network, as well as the risk of providing too much advanced warning to 
+ASIC developers. 
+
+It is further understood that this change will not prevent redesign of existing
+dedicated hardware with new ASIC chips. The intention of this change is only
+to disable currently active or mid-production hardware and provide time for
+POS development as well as larger algorithim changes to be well analyzed by 
+experts.
+
+The choice of FNV constants is made based on the formal specification at
+https://tools.ietf.org/html/draft-eastlake-fnv-14#section-2.1
+
+Typical ASIC synthesis tools would optimize multiplication of a constant
+in the FNV algorithm, reducing the area needed for the multiplier according
+to the hamming weight of the constant. To reduce the chance of ASIC adaptation
+through minor mask changes, we propose choosing new constants with a larger
+hamming weight, however care should be taken not to choose constants with too
+large of weight.
+
+The current FNV prime, 0x1000193 has a hamming weight of 6.   
+`HammingWeight(0x10001a7) = 7;`  
+`HammingWeight(0x10001ab) = 7;`    
+`HammingWeight(0x10001cf) = 8;`    
+`HammingWeight(0x10001e3) = 7;`    
+`HammingWeight(0x10001ef) = 9; // Not chosen`   
+`HammingWeight(0x10001f9) = 8;`  
+`HammingWeight(0x10001fb) = 9; // Not chosen`
+
+An exhaustive analysis was done regarding the dispersion of these constants as compared to 0x01000193.
+
+It can be empirically confirmed that no more than 1 duplicates occur in the 32 bit word space with these constants. 
```

Comment:
Do you have (a) reference(s) for these statements?

Reply:
Yes, I can provide the sample code I used to perform the exhaust analysis of the 32 bit word space if there is a good mechanism to do so.

Pseudo C Code

```c
uint32_t candidate = 0;
uint32_t dups = 0;
uint32_t fnv_candidate = 0x10001a7;
uint8_t *counts = malloc(0xFFFFFFFF);
memset(counts, '\0', 0xFFFFFFFF);
for ( candidate = 0; candidate <= 0xFFFFFFFF; candidate++) {
    result = (uint_32)(candidate * fnv_candidate);
    uint_8t oldCount = counts[result];
   counts[result] = counts[result]+1;
   if (oldCount != 0) {
      dups++;
   }
}
printf("Candidate 0x%08x : %i dups", dups);
```